dnsutil/util.go

```diff
@@ -0,0 +1,91 @@
+package dnsutil
+
+import "strings"
+
+// AddDomain adds origin to s if s is not already a FQDN.
+func AddDomain(origin, s string) string {
+	// If you want the result to end in a ".", pass an origin that ends in a "."
+	// ("origin.", "foo") -> "foo.origin."
+	// ("origin", "foo") -> "foo.origin"
+	// ("origin.", "foo.") -> "foo." (already a FQDN)
+	// ("origin.", "@") -> "origin." (@ represents the apex domain)
+	// ("origin.", "") -> "origin." (not obvious)
+
+	// These first cases are odd or borderline "shouldn't happen":
+	if len(origin) == 0 {
+		return s
+	}
+	if origin == "." && s == "@" {
+		return "@"
+	}
+
+	if s == "@" || len(s) == 0 {
+		return origin
+	}
+	if origin == "." {
+		if s[len(s)-1] == '.' {
+			return s
+		} else {
+			return s + "."
+		}
+	}
+	if s == "*" {
+		return "*." + origin
+	}
+	if s[len(s)-1] == '.' {
+		return s
+	}
+	return s + "." + origin
+}
+
+// TrimDomain trims the origin from a FQDN if it exists.
+func TrimDomain(origin, s string) string {
```

Comment:
TrimDomainName (also believe it's more natural to flip the order of the arguments)